app/controllers/web_hooks/slack_controller.rb

```diff
@@ -44,19 +44,18 @@ def allow_event?(event)
     Helios2Schema.subscriptions.trigger("eventPublished", {}, event)
   end
 
-  private def publish_announcement(event_callback)
-    announcement = Announcement.create!(parse_announcement(event_callback))
+  private def publish_announcement(event)
+    announcement = Announcement.create!(parse_announcement(event))
     Helios2Schema.subscriptions.trigger("announcementPublished", {}, announcement)
   end
 
-  private def parse_announcement(event_callback)
-    # Ex: "guests: Welcome to Roy and Amanda from Under Armour on May 23rd 2019 at 11:00 am in Providence"
-    type, content = event_callback[:event][:text].split(/: /)
-
+  private def parse_announcement(event)
+    # Ex: "<@U0LAN0Z89> guests= Welcome to Roy and Amanda from Under Armour on May 23rd 2019 at 11:00 am in Providence"
+    _bot_id, type, content = event[:text].split(/>|: /).map(&:strip)
```

Comment:
I'm a little worried this may match other parts of the string. You can use a more exact regex to avoid that:
```
if match = event[:text].match(/^<\S+> (\S+) (.*)$/)
  type, content = match.captures
  # stuffs
else
  # potential help message
end
```